rubin_sim/scheduler/surveys/scripted_surveys.py

```diff
@@ -53,30 +65,13 @@ def add_observation(self, observation, indx=None, **kwargs):
                     detailer.add_observation(observation, **kwargs)
                 self.reward_checked = False
 
-                # was it taken in the right time window, and hasn't already been marked as observed.
-                time_matches = np.where(
-                    (observation["mjd"] > self.mjd_start)
-                    & (observation["mjd"] < self.obs_wanted["flush_by_mjd"])
-                    & (~self.obs_wanted["observed"])
-                    & (observation["note"] == self.obs_wanted["note"])
-                )[0]
-
-                for match in time_matches:
-                    distances = _angularSeparation(
-                        self.obs_wanted[match]["RA"],
-                        self.obs_wanted[match]["dec"],
-                        observation["RA"],
-                        observation["dec"],
-                    )
-                    if (distances < self.obs_wanted[match]["dist_tol"]) & (
-                        self.obs_wanted[match]["filter"] == observation["filter"]
-                    ):
-                        # Log it as observed.
-                        self.obs_wanted["observed"][match] = True
-                        self.scheduled_obs = self.obs_wanted["mjd"][
-                            ~self.obs_wanted["observed"]
-                        ]
-                        break
+                # find the index 
+                indx = np.searchsorted(self.obs_wanted['scripted_id'], observation['scripted_id'])
+                # If it matches scripted_id and note, mark it as observed and update scheduled observation list.
+                if self.obs_wanted['scripted_id'][indx] == observation['scripted_id']:
```

Comment:
I think you can merge these 2 `if` statements into one no?

```
if self.obs_wanted['scripted_id'][indx] == observation['scripted_id'] and self.obs_wanted['note'][indx] == observation['note']:
...
```

Reply:
Yeah, I'd like to deprecate `field_id` and `survey_id` (and `block_id` can be handy, but isn't critical).